### PR TITLE
[Serving] Use before param in add_aggregate function as expected.

### PR DESCRIPTION
**Pull request by davesh0812**
Fix bug https://jira.iguazeng.com/browse/ML-2346

### CODE REVIEW

File: mlrun/feature_store/feature_set.py [hunk 1]
```diff
@@ -717,10 +717,12 @@ def upsert_feature(name):
         else:
             class_args = {}
             self._aggregations[aggregation["name"]] = aggregation
+            if before is None and after is None:
+                after = previous_step
             if not self.spec.engine or self.spec.engine == "storey":
                 step = graph.add_step(
                     name=step_name,
-                    after=after or previous_step,
```
Comment (yaronha) [inline mlrun/feature_store/feature_set.py:723]: i see u kept `previous_step` in the spark below, shouldn't it be the same?

File: mlrun/serving/states.py [hunk 2]
```diff
@@ -783,6 +783,7 @@ def insert_step(self, key, step, after, before=None):
                 raise GraphError(
                     f"graph loop, step {before} is specified in before and/or after {key}"
                 )
+            self[step.name].after_step(*self[before].after, append=False)
```
Comment (yaronha) [inline mlrun/serving/states.py:786]: what issue is it solving?

Comment (davesh0812) [reply mlrun/serving/states.py:786]: Correctly defines the position of the new step in the graph. 
A step contains only the after parameter, so if we don't define it here, the step we entered will be disconnected from the beginning of the graph.

File: mlrun/feature_store/feature_set.py (hunk 1 above)
Comment (yaronha) [reply mlrun/feature_store/feature_set.py:723]: so i guess `previous_step` is not used anywhere?

**Review by yaronha — APPROVED** [review]

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**